app/code/Magento/Sales/Model/Order/Shipment.php

```diff
@@ -435,19 +463,28 @@ public function addComment($comment, $notify = false, $visibleOnFront = false)
      */
     public function getCommentsCollection($reload = false)
     {
-        if (!$this->hasData(ShipmentInterface::COMMENTS) || $reload) {
+        if (!$this->hasData(self::COMMENTS_COLLECTION) || $reload) {
             $comments = $this->_commentCollectionFactory->create()
                 ->setShipmentFilter($this->getId())
                 ->setCreatedAtOrder();
-            $this->setComments($comments);
+            $this->setCommentsCollection($comments);
 
             if ($this->getId()) {
-                foreach ($this->getComments() as $comment) {
+                foreach ($this->getCommentsCollection() as $comment) {
                     $comment->setShipment($this);
                 }
             }
         }
-        return $this->getComments();
+        return $this->getData(self::COMMENTS_COLLECTION);
+    }
+
+    /**
+     * @param \Magento\Sales\Model\ResourceModel\Order\Shipment\Comment\Collection $comments
+     * @return $this
+     */
+    public function setCommentsCollection($comments)
```

Comment:
You can't add public methods to the class marked as @api, this is BIC (Backward Incompatible Changes).

Reply:
It is backward compatible change as adding new function do not change existing behaviour and results.
If it is some sort of policy, then can you please send me a link to it?

Reply:
Actually not, please, see https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/ - **Introduction of a method to a class or interface** section.